compose-guard/gradle-plugin/src/functionalTest/kotlin/com/joetr/compose/guard/task/ComposeCompilerReportCheckTaskTest.kt

```diff
@@ -345,4 +345,58 @@ class ComposeCompilerReportCheckTaskTest {
         assertThat(secondaryCheckResult.output).contains("<runtime stability> = Unstable")
         assertThat(secondaryCheckResult).task(checkTask).failed()
     }
+
+    @Test
+    fun `can detect when new unstable parameters are added`() {
+        val project = BasicAndroidProject.getComposeProject()
+        val generateTask = ":android:releaseComposeCompilerGenerate"
+
+        // add unstable class and dynamic property so that check doesn't fail and generate golden metrics
+        project.projectDir("android").resolve("src/main/kotlin/com/example/myapplication/TestComposable.kt").toFile().writeText(
+            """
+            package com.example.myapplication
+
+            import androidx.compose.material3.Text
+            import androidx.compose.runtime.Composable
+
+            data class UnstableClass(var value: String)
+            data class OtherUnstableClass(var value: String)
+
+            @Composable
+            fun TestComposable(firstUnstable: UnstableClass) {
+                Text(text = firstUnstable.value)
+            }
+            """.trimIndent(),
+        )
+        // Generate golden metrics with the new class included
+        var generateResult = project.execute(generateTask)
+        assertThat(generateResult).task(generateTask).succeeded()
+
+        // modify the composable to introduce a new unstable parameter
+        project.projectDir("android").resolve("src/main/kotlin/com/example/myapplication/TestComposable.kt").toFile().writeText(
+            """
+            package com.example.myapplication
+
+            import androidx.compose.material3.Text
+            import androidx.compose.runtime.Composable
+
+            data class UnstableClass(var value: String)
+            data class OtherUnstableClass(var value: String)
+
+            @Composable
+            fun TestComposable(firstUnstable: UnstableClass, newUnstable: OtherUnstableClass) {
```

Comment:
can you either add another test or modify the existing one to delete an existing unstable class too and verifies it still triggers? we still want to make sure it triggers if not just the absolute amount of unstable classes increase

Reply:
I updated the check and the test so now it's actually checking if any new unstable param was added, even if it's replacing the old unstable param. This was absolute amount really doesn't matter. I think that satisfies what you're talking aboout, right?

Reply:
absolutely